OpenRA.Mods.Common/Traits/Conditions/ToggleConditionOnOrder.cs

```diff
@@ -103,5 +103,12 @@ protected override void TraitResumed(Actor self)
 			// Unpausing the trait restores the previous state
 			SetCondition(self, enabled);
 		}
+
+		void INotifyOwnerChanged.OnOwnerChanged(Actor self, Player oldOwner, Player newOwner)
+		{
+			// Switch toggle back to untoggled if ownerchanged
+			enabled = false;
+			SetCondition(self, false);
+		}
```

Comment:
Won't the condition be removed from `TraitDisabled`? Or does removing an actor from the world not trigger `TraitDisabled`?

Reply:
The owner change is to solve a problem that AI-captured building which has already been in power down state won't be able to controlled by this function since we only restore the power consumption in cached list. 

> Capturing isn't enabled in most mods, and in any case we should really be restoring power to a building if it changes owner. Fixing this would make the code here much simpler.

by @pchote 

So I just did that.

Reply:
And yeah we can replace 

```
			enabled = false;
			SetCondition(self, false);
```

as

```
TraitDisabled(self)
```

Reply:
No, I was more wondering if this doesn't make more sense to be called from `INotifyRemovedFromWorld` or if that already calls `TraitDisabled` (which it apparently doesn't).

Reply:
Ummm... 
Then I don't know that? I am not familiar with this interface.

Reply:
Ummm... 
Then I don't know that? I am not familiar with this interface.